wallet/udb/txmined.go

```diff
@@ -3405,96 +3477,51 @@ func (s *Store) balanceFullScan(ns, addrmgrNs walletdb.ReadBucket, minConf int32
-			serializedTx := extractRawUnminedTx(rawUnmined)
-			var rec TxRecord
-			err = rec.MsgTx.Deserialize(bytes.NewReader(serializedTx))
-			if err != nil {
-				return nil, err
-			}
-			votingAuthorityAmt := dcrutil.Amount(0)
-			lockedByTicketsAmt := dcrutil.Amount(0)
-
-			// Calculate total input amount which will allow a proper fee calculation.
-			// Fee is needed to be removed from the stake.AmountFromSStxPkScrCommitment
-			// due to the way tickets are constructed and rewards are calculated.
-			totalInputAmount := dcrutil.Amount(0)
-			for _, txin := range rec.MsgTx.TxIn {
-				rawUnmined := existsRawUnmined(ns, txin.PreviousOutPoint.Hash[:])
-				if rawUnmined != nil {
-					serializedTx := extractRawUnminedTx(rawUnmined)
-					var tx wire.MsgTx
-					err = tx.Deserialize(bytes.NewReader(serializedTx))
-					if err != nil {
-						return nil, err
-					}
-					if int(txin.PreviousOutPoint.Index) < len(tx.TxOut) {
-						totalInputAmount += dcrutil.Amount(tx.TxOut[txin.PreviousOutPoint.Index].Value)
-					}
-				} else {
-					_, txVal := latestTxRecord(ns, txin.PreviousOutPoint.Hash[:])
-					if txVal == nil {
-						continue
-					}
-					var tx wire.MsgTx
-					err = readRawTxRecordMsgTx(&txin.PreviousOutPoint.Hash, txVal, &tx)
-					if err != nil {
-						return nil, err
-					}
-					if int(txin.PreviousOutPoint.Index) < len(tx.TxOut) {
-						totalInputAmount += dcrutil.Amount(tx.TxOut[txin.PreviousOutPoint.Index].Value)
-					}
-				}
-			}
-			for i, txout := range rec.MsgTx.TxOut {
-				if i%2 != 0 {
-					addr, err := stake.AddrFromSStxPkScrCommitment(txout.PkScript,
-						s.chainParams)
-					if err != nil {
-						return nil, err
-					}
-					amt, err := stake.AmountFromSStxPkScrCommitment(txout.PkScript)
-					if err != nil {
-						return nil, err
-					}
-					votingAuthorityAmt += amt
-					if _, err := s.acctLookupFunc(addrmgrNs, addr); err != nil {
-						if errors.Is(errors.NotExist, err) {
-							continue
-						}
-						return nil, err
-					}
-					lockedByTicketsAmt += amt
-				}
-			}
-			// Calculate the fee here due to the commitamt in the OP_SSTX output script being the total
-			// value in.
-			fee := dcrutil.Amount(0)
-			if totalInputAmount > 0 {
-				fee = totalInputAmount - utxoAmt
-			}
-			if lockedByTicketsAmt > 0 {
-				// Only calculate proper lockedbyticketstamt if > 0
-				ab.LockedByTickets += lockedByTicketsAmt - fee
-			}
-			ab.VotingAuthority += votingAuthorityAmt - fee
+			ab.VotingAuthority += utxoAmt
 		case txscript.OP_SSGEN:
 			fallthrough
 		case txscript.OP_SSRTX:
 			ab.ImmatureStakeGeneration += utxoAmt
+			ab.Total += utxoAmt
 		case txscript.OP_SSTXCHANGE:
+			ab.Total += utxoAmt
 			continue
 		default:
 			log.Warnf("Unhandled unconfirmed opcode %v: %v", opcode, v)
 		}
 	}
 
+	// Account for ticket commitments by iterating over the unspent commitments
+	// index.
+	it := makeUnspentTicketCommitsIterator(ns)
+	for it.next() {
+		// TODO: Semantics of minconf for commitments?
```

Comment:
I'd just ignore it.  They aren't spendable outputs after all, so their chain depth doesn't matter.